### ISSUE TITLE
Core test CopyNodeFromCSVTest.CopyNodeTest fails due to incorrect output on all platforms on GitHub-hosted runners

### ISSUE THREAD

**Issue by russell-liu** [kuzudb/kuzu#2059]
When running the core tests (`make test`) on commit d3ed7f26ee2575fc2e702fd3d495405ee869555a, all tests pass; on the successive commit 9fe713ebd1199f1fdc6614d818d1a7ad951a9edb and commits thereafter, the test CopyNodeFromCSVTest.CopyNodeTest fails.
Example, on a GitHub-hosted runner running a dockerized Ubuntu 23.04 container (I have combined all steps into one to make the setup as closely reproducible locally as possible; all tests pass when d3ed7f26 is checked out but the one test fails when 9fe713eb is checked out):
https://github.com/kuzudb/kuzu/actions/runs/6202502792/job/16841357639
However, on a local dockerized Ubuntu 23.04 container, running the exact same commands in the workflow file (https://github.com/kuzudb/kuzu/actions/runs/6202502792/workflow) results in all test passing in both instances. The container was also configured to have the same specification as given here: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources. I am unsure what other differences could be causing different behavior on a GitHub-hosted vs. local setup.
The CopyNodeFromCSVTest.CopyNodeTest test has also failed at various commits including and after 9fe713eb, on both GitHub-provided and dockerized runners, as long as they are being hosted by GitHub. Here are some examples:
windows-2019: https://github.com/kuzudb/kuzu/actions/runs/6165580879/job/16733589715
macos-12: https://github.com/kuzudb/kuzu/actions/runs/6165636640/job/16733756623
ubuntu-20.04: https://github.com/kuzudb/kuzu/actions/runs/6177200943/job/16767827598
ubuntu-22.04: https://github.com/kuzudb/kuzu/actions/runs/6189924193/job/16805055995
There are more, but I forgot to keep track of them. I could dig them up if necessary.
Note: the test failure message varies depending on which commit including or after 9fe713eb is checked out, but it is the same test that fails. It has something to do with this line:
```
49992|50|31.582059|False|1551-07-19|1551-07-19 16:28:31||[5,67,66]
```
On the other hand, the test has never been demonstrated to fail on machines not hosted by GitHub. For instance, the commit passed all tests on the self-hosted machine when it was made (https://github.com/kuzudb/kuzu/actions/runs/6114565146/job/16596540428), and it has also passed all tests on various Linux distro containers and one Ubuntu 20.04 virtual machine by my testing.

**Comment by russell-liu** [comment]
The test still fails on a GitHub-hosted runner when run with minimal concurrency (`NUM_THREADS=1 TEST_JOBS=1`):
https://github.com/kuzudb/kuzu/actions/runs/6251618172/job/16973114491

**Comment by russell-liu** [comment]
Update: the test no longer fails on any platform as of commit f8d48a62ebbb83b066a8942e6c3bf93411441241, so it was somehow fixed somewhere between 9fe713ebd1199f1fdc6614d818d1a7ad951a9edb and that.
